packages/mds-policy-service/service/provider.ts

```diff
@@ -103,8 +113,30 @@ export const PolicyServiceProvider: ServiceProvider<PolicyService, PolicyService
     }),
   writePolicyIntentToPolicy: (context, intent_draft) =>
     serviceErrorWrapper('writePolicyIntentToPolicy', async () => {
-      // TODO insert ajv validation here
-      const { policy_id } = await PolicyRepository.writePolicy(translateIntentToPolicy(intent_draft))
+      switch (intent_draft.intent_type) {
+        case 'no_parking': {
+          validateNoParkingPolicyIntent(intent_draft)
+          break
+        }
+        case 'permitted_vehicle_count': {
+          validatePermittedVehicleCountPolicyIntent(intent_draft)
+          break
+        }
+        case 'parking_time_limit': {
+          validateParkingTimeLimitPolicyIntent(intent_draft)
+          break
+        }
+        case 'permitted_parking': {
+          validatePermittedParkingPolicyIntent(intent_draft)
+          break
+        }
+        default: {
+          throw new BadParamsError('Unknown intent type!')
+        }
+      }
```

Comment:
My slight style preference would be to use an object lookup here instead of a switch so we could have the lookup be exhaustive (and in the future once new intents are added, they would *have* to be explicitly handled, versus implicitly handled by the `default`). Something like:

```
const intentTypeToValidator = {
  'no_parking': validateNoParkingPolicyIntent,
  ...
}

intentTypeToValidator[intent_draft.intent_type](intent_draft)
```